src/main/java/com/microsoft/sqlserver/jdbc/SQLServerDatabaseMetaData.java

```diff
@@ -601,72 +601,63 @@ private static String EscapeIDName(String inID) throws SQLServerException {
-    // to be inline with JDBC spec
-    private static final String[] getColumnsColumnNamesKatmai = {/* 1 */ TABLE_CAT, /* 2 */ TABLE_SCHEM,
-            /* 3 */ TABLE_NAME, /* 4 */ COLUMN_NAME, /* 5 */ DATA_TYPE, /* 6 */ TYPE_NAME, /* 7 */ COLUMN_SIZE,
-            /* 8 */ BUFFER_LENGTH, /* 9 */ DECIMAL_DIGITS, /* 10 */ NUM_PREC_RADIX, /* 11 */ NULLABLE, /* 12 */ REMARKS,
-            /* 13 */ COLUMN_DEF, /* 14 */ SQL_DATA_TYPE, /* 15 */ SQL_DATETIME_SUB, /* 16 */ CHAR_OCTET_LENGTH,
-            /* 17 */ ORDINAL_POSITION, /* 18 */ IS_NULLABLE, /* 20 */ SS_IS_SPARSE, /* 20 */ SS_IS_COLUMN_SET,
-            /* 21 */ IS_GENERATEDCOLUMN, /* 22 */ IS_AUTOINCREMENT};
-
     @Override
     public java.sql.ResultSet getColumns(String catalog, String schema, String table,
-            String col) throws SQLServerException, SQLTimeoutException {
+            String col) throws SQLException, SQLTimeoutException {
         if (loggerExternal.isLoggable(Level.FINER) && Util.IsActivityTraceOn()) {
             loggerExternal.finer(toString() + " ActivityId: " + ActivityCorrelator.getNext().toString());
         }
         checkClosed();
-
-        // sp_columns supports wild carding schema table and columns
-        String column = EscapeIDName(col);
-        table = EscapeIDName(table);
-        schema = EscapeIDName(schema);
-
-        /*
-         * sp_columns [ @table_name = ] object [ , [ @table_owner = ] owner ] [ , [ @table_qualifier = ] qualifier ] [ ,
-         * [ @column_name = ] column ] [ , [ @ODBCVer = ] ODBCVer ]
-         */
-        String[] arguments;
-        if (connection.isKatmaiOrLater())
-            arguments = new String[6];
-        else
-            arguments = new String[5];
-        arguments[0] = table;
-        arguments[1] = schema;
-        arguments[2] = catalog;
-        arguments[3] = column;
-        if (connection.isKatmaiOrLater()) {
-            arguments[4] = "2"; // give information about everything including
-                                // sparse columns
-            arguments[5] = "3"; // odbc version
-        } else
-            arguments[4] = "3"; // odbc version
-        SQLServerResultSet rs;
-        if (connection.isKatmaiOrLater())
-            rs = getResultSetWithProvidedColumnNames(catalog, CallableHandles.SP_COLUMNS, arguments,
-                    getColumnsColumnNamesKatmai);
-        else
-            rs = getResultSetWithProvidedColumnNames(catalog, CallableHandles.SP_COLUMNS, arguments,
-                    getColumnsColumnNames);
-        // Hook in a filter on the DATA_TYPE column of the result set we're
-        // going to return that converts the ODBC values from sp_columns
-        // into JDBC values.
-        rs.getColumn(5).setFilter(new DataTypeFilter());
-
-        if (connection.isKatmaiOrLater()) {
-            rs.getColumn(22).setFilter(new IntColumnIdentityFilter());
+        String originalCatalog = null;
+        originalCatalog = switchCatalogs(catalog);
```

Comment:
This won't work against Azure.

Reply:
Original implementation did not take Azure into account either, method will just throw w/e error Azure throws when attempting to switch cats. Switchcatalog returns null immediately if no catalog is provided.